packages/firestore/src/lite-api/query.ts

```diff
@@ -858,3 +989,15 @@ function validateOrderByAndInequalityMatch(
     );
   }
 }
+
+export function validateQueryFilterConstraint(
```

Comment:
As you mentioned last time, it's not great that we need to do this validation, and points to room for improvement in the API.

Reply:
I'm going to leave this in until we get approval on a different API that doesn't need this. However, I realized that I need to test this validation.

Reply:
👍

Reply:
Given that `or` and `and` are currently not public, and that we expect this validateQueryFilterConstraint function to hopefully go away with the new API design, I think you can skip adding tests for it right now.